lib/iris/common/metadata.py

```diff
@@ -1438,14 +1413,6 @@ def values(self):
         emsg = "Require a subclass of {!r}, got {!r}."
         raise TypeError(emsg.format(BaseMetadata.__name__, cls))
 
-    # Check whether kwargs have valid fields for the specified metadata.
-    if kwargs:
-        extra = [field for field in kwargs.keys() if field not in cls._fields]
-        if extra:
-            bad = ", ".join(map(lambda field: "{!r}".format(field), extra))
-            emsg = "Invalid {!r} field parameters, got {}."
-            raise ValueError(emsg.format(cls.__name__, bad))
-
```

Comment:
Avoid checking the `kwargs` at metadata manager instance creation time, and instead do it within the `metadata_manager_factory`. This also deftly avoids the potential issue of non hashable `kwargs` being passed through `lru_cache`.

Reply:
Not sure I understand this comment - from the code above it looks like the lru_cache is decorating a function of single argument `cls`.  How does the `kwargs` affect the cache?  Is there some closure I'm not spotting?

Reply:
Ah, are you saying that you've moved `kwargs` to `metadata_manager_factory`, so it never sees the cache?

Reply:
Yeah, it's a bit subtle, but there's no need to pass the `kwargs` into `_factory_cache`. I wanted to avoid potentially passing in non-hashable `kwargs` (as `lru_cache` enforces this), plus I also want things to fail fast i.e., within the factory and not the cache. Does that help?